service/config_provider.go

```diff
@@ -75,6 +75,37 @@ type configProvider struct {
 	watcher chan error
 }
 
+// ConfigProviderOptions has options that change the behavior of ConfigProvider
+// returned by NewConfigProvider()
+type ConfigProviderOptions struct {
+	configMapProviders map[string]configmapprovider.Provider
+	cfgMapConverters   []config.MapConverterFunc
+	configUnmarshaler  configunmarshaler.ConfigUnmarshaler
+}
+
+// ConfigProviderOption is an option to change the behavior of ConfigProvider
+// returned by NewConfigProvider()
+type ConfigProviderOption func(opts *ConfigProviderOptions)
+
+func WithConfigMapProviders(c map[string]configmapprovider.Provider) ConfigProviderOption {
+	return func(opts *ConfigProviderOptions) {
+		opts.configMapProviders = c
+	}
+}
+
+func WithCfgMapConverters(c []config.MapConverterFunc) ConfigProviderOption {
+	return func(opts *ConfigProviderOptions) {
+		opts.cfgMapConverters = c
+	}
+}
```

Comment:
The order of converters matters, the question I have is: do we ask the user to give a slice of them or one by one:

```golang

func WithCfgMapConverters(c config.MapConverterFunc) ConfigProviderOption {
	return func(opts *ConfigProviderOptions) {
		opts.cfgMapConverters = append(opts.cfgMapConverters, c)
	}
}

return NewConfigProvider(
	configLocations,
	WithCfgMapConverter(configmapprovider.NewOverwritePropertiesConverter(properties)),
	WithCfgMapConverter(configmapprovider.NewExpandConverter())
	)
```

Reply:
I think this is not a good idea, since if we define a default "converter" user cannot remove/change that.